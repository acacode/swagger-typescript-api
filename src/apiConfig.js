const _ = require("lodash");

const createApiConfig = (swaggerSchema) => {
  const { info, servers, host, basePath, externalDocs, tags } = swaggerSchema;
  const server = (servers && servers[0]) || { url: "" };
  const { title = "No title", version, description: schemaDescription = "" } = info || {};
  const { url: serverUrl } = server;

  return {
    info: info || {},
    servers: servers || [],
    basePath,
    host,
    externalDocs: _.merge(
      {
        url: "",
        description: "",
      },
      externalDocs,
    ),
    tags: _.compact(tags),
    baseUrl: serverUrl,
    title,
    version,
  };
};

module.exports = {
  createApiConfig,
};
