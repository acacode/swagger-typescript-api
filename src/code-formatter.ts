import * as path from "node:path";
import { Biome, Distribution } from "@biomejs/js-api";
import * as nanoid from "nanoid";
import * as typescript from "typescript";
import type { CodeGenConfig } from "./configuration.js";

export class CodeFormatter {
  config: CodeGenConfig;

  constructor(config: CodeGenConfig) {
    this.config = config;
  }

  removeUnusedImports = (content: string) => {
    const tempFileName = "file.ts";

    const host = new TsLanguageServiceHost(tempFileName, content);
    const languageService = typescript.createLanguageService(host);

    const fileTextChanges = languageService.organizeImports(
      { type: "file", fileName: tempFileName },
      { newLineCharacter: typescript.sys.newLine },
      undefined,
    )[0];

    if (fileTextChanges?.textChanges.length) {
      return fileTextChanges.textChanges.reduceRight(
        (content, { span, newText }) =>
          `${content.slice(0, span.start)}${newText}${content.slice(
            span.start + span.length,
          )}`,
        content,
      );
    }

    return content;
  };

  format = async (content: string) => {
    const biome = await Biome.create({ distribution: Distribution.NODE });
    const biomeProject = biome.openProject();
    biome.applyConfiguration(biomeProject.projectKey, {
      files: { maxSize: Number.MAX_SAFE_INTEGER },
      formatter: { indentStyle: "space" },
    });
    const formatted = biome.formatContent(biomeProject.projectKey, content, {
      filePath: path.format({ name: nanoid.nanoid(), ext: ".ts" }),
    });
    return formatted.content;
  };

  formatCode = async (
    code: string,
    { removeUnusedImports = true, format = true } = {},
  ) => {
    if (removeUnusedImports) {
      code = this.removeUnusedImports(code);
    }
    if (format) {
      code = await this.format(code);
    }
    return code;
  };
}

class TsLanguageServiceHost {
  fileName: string;
  content: string;
  compilerOptions: typescript.CompilerOptions;

  constructor(fileName: string, content: string) {
    this.fileName = fileName;
    this.content = content;
    const tsconfig = typescript.findConfigFile(
      fileName,
      typescript.sys.fileExists,
    );
    this.compilerOptions = tsconfig
      ? typescript.convertCompilerOptionsFromJson(
          typescript.readConfigFile(tsconfig, typescript.sys.readFile).config
            .compilerOptions,
          "",
        ).options
      : typescript.getDefaultCompilerOptions();
  }

  getNewLine() {
    return "newLine" in typescript.sys ? typescript.sys.newLine : "\n";
  }
  getScriptFileNames() {
    return [this.fileName];
  }
  getCompilationSettings() {
    return this.compilerOptions;
  }
  getDefaultLibFileName() {
    return typescript.getDefaultLibFileName(this.getCompilationSettings());
  }
  getCurrentDirectory() {
    return process.cwd();
  }
  getScriptVersion() {
    return typescript.version;
  }
  getScriptSnapshot() {
    return typescript.ScriptSnapshot.fromString(this.content);
  }
  readFile(fileName: string, encoding: string) {
    if (fileName === this.fileName) {
      return this.content;
    }

    return typescript.sys.readFile(fileName, encoding);
  }
  fileExists(path: string) {
    return typescript.sys.fileExists(path);
  }
}
