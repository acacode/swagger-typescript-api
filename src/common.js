const _ = require("lodash");

module.exports = {
  formatDescription: (description, inline) => {
    if (!description) return "";

    let prettified = description;

    prettified = _.replace(prettified, /\*\//g, "*/");

    const hasMultipleLines = _.includes(prettified, "\n");

    if (!hasMultipleLines) return prettified;

    if (inline) {
      return _(prettified)
        .split(/\n/g)
        .map((part) => _.trim(part))
        .compact()
        .join(" ")
        .valueOf();
    }

    return _.replace(prettified, /\n$/g, "");
  },
  internalCase: (value) => _.camelCase(_.lowerCase(value)),
  classNameCase: (value) => _.upperFirst(_.camelCase(value)),
};
