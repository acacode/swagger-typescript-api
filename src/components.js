const _ = require("lodash");
const { parseSchema } = require("./schema");
const { addToConfig } = require("./config");

/**
 * 
 * @typedef TypeInfo
 * {
 *    typeName: "Foo",
 *    componentName: "schemas",
 *    rawTypeData: {...},
 *    typeData: {...} (result parseSchema())
 * } 
 */

/**
 * @returns {{ "#/components/schemas/Foo": TypeInfo, ... }}
 */
const createComponentsMap = components => {
  const componentsMap = _.reduce(components, (map, component, componentName) => {
    _.each(component, (rawTypeData, typeName) => {
      // only map data for now
      map[`#/components/${componentName}/${typeName}`] = {
        typeName,
        rawTypeData,
        componentName,
        typeData: null,
      }
    })
    return map;
  }, {})

  addToConfig({ componentsMap })

  return componentsMap;
}



/**
 * @returns {TypeInfo[]}
 */
const filterComponentsMap = (componentsMap, componentName) =>
  _.filter(componentsMap, (v, ref) => _.startsWith(ref, `#/components/${componentName}`))


/** @returns {{ type, typeIdentifier, name, description, content }} */
const getTypeData = typeInfo => {
  if (!typeInfo.typeData) {
    typeInfo.typeData = parseSchema(typeInfo.rawTypeData, typeInfo.typeName)
  }

  return typeInfo.typeData;
}

module.exports = {
  getTypeData,
  createComponentsMap,
  filterComponentsMap,
}