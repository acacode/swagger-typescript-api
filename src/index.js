#!/usr/bin/env node

// Copyright (c) 2019-present acacode
// Node module: swagger-typescript-api
// This file is licensed under the MIT License.
// License text available at https://opensource.org/licenses/MIT
// Repository https://github.com/acacode/swagger-typescript-api

const _ = require("lodash");
const { parseSchemas } = require("./schema");
const { parseRoutes, groupRoutes } = require("./routes");
const { createApiConfig } = require("./apiConfig");
const { prepareModelType } = require("./modelTypes");
const { getSwaggerObject, fixSwaggerScheme, convertSwaggerObject } = require("./swagger");
const { createComponentsMap, filterComponentsMap } = require("./components");
const { createFile, pathIsExist, pathIsDir, createDir, cleanDir } = require("./files");
const { addToConfig, config } = require("./config");
const { getTemplates, getTemplatePaths, renderTemplate, getTemplate } = require("./templates");
const constants = require("./constants");
const { generateOutputFiles } = require("./output");
const formatFileContent = require("./formatFileContent");
const { logger } = require("./logger");
const { ComponentTypeNameResolver } = require("./utils/resolveName");
const { getPrettierOptions } = require("./prettierOptions");

module.exports = {
  constants: constants,
  generateApi: ({
    input,
    output,
    url,
    spec,
    name: fileName,
    toJS: translateToJavaScript = config.toJS,
    modular,
    templates,
    generateResponses = config.generateResponses,
    defaultResponseAsSuccess = config.defaultResponseAsSuccess,
    generateRouteTypes = config.generateRouteTypes,
    generateClient = config.generateClient,
    httpClientType = config.httpClientType,
    generateUnionEnums = config.generateUnionEnums,
    moduleNameIndex = config.moduleNameIndex,
    moduleNameFirstTag = config.moduleNameFirstTag,
    extractRequestParams = config.extractRequestParams,
    extractRequestBody = config.extractRequestBody,
    defaultResponseType = config.defaultResponseType,
    unwrapResponseData = config.unwrapResponseData,
    disableThrowOnError = config.disableThrowOnError,
    sortTypes = config.sortTypes,
    singleHttpClient = config.singleHttpClient,
    prettier: prettierOptions = getPrettierOptions(),
    hooks: rawHooks,
    extraTemplates,
    enumNamesAsValues,
    disableStrictSSL = config.disableStrictSSL,
    disableProxy = config.disableProxy,
    cleanOutput,
    silent = config.silent,
    typePrefix = config.typePrefix,
    typeSuffix = config.typeSuffix,
    patch = config.patch,
  }) =>
    new Promise((resolve, reject) => {
      addToConfig({
        defaultResponseAsSuccess,
        generateRouteTypes,
        generateClient,
        httpClientType,
        generateResponses,
        templates,
        generateUnionEnums,
        moduleNameIndex,
        moduleNameFirstTag,
        prettierOptions,
        modular,
        extractRequestParams,
        extractRequestBody,
        hooks: _.merge(config.hooks, rawHooks || {}),
        enumNamesAsValues,
        disableStrictSSL,
        disableProxy,
        cleanOutput,
        defaultResponseType,
        unwrapResponseData,
        disableThrowOnError,
        sortTypes,
        singleHttpClient,
        constants,
        silent,
        toJS: translateToJavaScript,
        typePrefix,
        typeSuffix,
        patch,
      });
      (spec
        ? convertSwaggerObject(spec, { patch })
        : getSwaggerObject(input, url, disableStrictSSL, disableProxy, { patch })
      )
        .then(({ usageSchema, originalSchema }) => {
          const templatePaths = getTemplatePaths(config);

          addToConfig({ templatePaths });

          const templatesToRender = getTemplates(config);

          logger.event("start generating your typescript api");

          fixSwaggerScheme(usageSchema, originalSchema);

          addToConfig({
            swaggerSchema: usageSchema,
            originalSchema,
            templatesToRender,
          });

          const { components } = usageSchema;

          addToConfig(config.hooks.onInit(config) || config);

          const componentsMap = createComponentsMap(components);

          const componentSchemasNames = filterComponentsMap(componentsMap, "schemas").map(
            (c) => c.typeName,
          );

          addToConfig({
            componentTypeNameResolver: new ComponentTypeNameResolver(componentSchemasNames),
          });

          const parsedSchemas = parseSchemas(components);

          config.routeNameDuplicatesMap.clear();

          const routes = parseRoutes({
            usageSchema,
            parsedSchemas,
            moduleNameIndex,
            moduleNameFirstTag,
            extractRequestParams,
          });

          const hasSecurityRoutes = routes.some((route) => route.security);
          const hasQueryRoutes = routes.some((route) => route.hasQuery);
          const hasFormDataRoutes = routes.some((route) => route.hasFormDataParams);

          const usageComponentSchemas = filterComponentsMap(componentsMap, "schemas");
          const sortByProperty = (o1, o2, propertyName) => {
            if(o1[propertyName] > o2[propertyName]) {
              return 1;
            }
            if(o1[propertyName] < o2[propertyName]) {
              return -1;
            }
            return 0;
          }
          const sortByTypeName = (o1, o2) => sortByProperty(o1, o2, 'typeName');

          const sortByName = (o1, o2) => sortByProperty(o1, o2, 'name');

          const sortSchemas = (schemas) => {
            if(config.sortTypes) {
              return schemas.sort(sortByTypeName).map((schema) => {
                if(schema.rawTypeData?.properties) {
                  return {
                    ...schema,
                    rawTypeData: {
                      ...schema.rawTypeData,
                      '$parsed': {...schema.rawTypeData['$parsed'], content: schema.rawTypeData['$parsed'].content.sort(sortByName)}
                    }
                  }
                }
                return schema;
              });
            }
            return schemas;
          };

          const rawConfiguration = {
            apiConfig: createApiConfig(usageSchema),
            config,
            modelTypes: _.map(sortSchemas(usageComponentSchemas), prepareModelType),
            rawModelTypes: usageComponentSchemas,
            hasFormDataRoutes,
            hasSecurityRoutes,
            hasQueryRoutes,
            generateResponses,
            routes: groupRoutes(routes),
            extraTemplates,
            fileName,
            translateToJavaScript,
            utils: {
              ...require("./render/utils"),
              ...require("./common"),
            },
          };

          const configuration = config.hooks.onPrepareConfig(rawConfiguration) || rawConfiguration;

          if (pathIsExist(output)) {
            if (cleanOutput) {
              cleanDir(output);
            }
          } else {
            createDir(output);
          }

          const files = generateOutputFiles({
            modular,
            templatesToRender,
            configuration,
          });

          const isDirPath = pathIsDir(output);

          const generatedFiles = files.map((file) => {
            if (!isDirPath) return file;

            if (translateToJavaScript) {
              createFile({
                path: output,
                fileName: file.name,
                content: file.content,
                withPrefix: true,
              });
              createFile({
                path: output,
                fileName: file.declaration.name,
                content: file.declaration.content,
                withPrefix: true,
              });
              logger.success(`javascript api file`, file.name, `created in ${output}`);
            } else {
              createFile({
                path: output,
                fileName: file.name,
                content: file.content,
                withPrefix: true,
              });
              logger.success(`typescript api file`, file.name, `created in ${output}`);
            }

            return file;
          });

          resolve({
            files: generatedFiles,
            configuration,
            getTemplate,
            renderTemplate,
            createFile,
            formatTSContent: formatFileContent,
          });
        })
        .catch((e) => {
          reject(e);
          throw new Error("Swagger schema parse error!\r\n " + e);
        });
    }),
};
