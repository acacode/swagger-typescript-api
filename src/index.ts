import { consola } from "consola";
import type { GenerateApiConfiguration } from "../types/index.js";
import { CodeGenProcess } from "./code-gen-process.js";

export async function generateApi(
  config: Partial<GenerateApiConfiguration["config"]>,
) {
  if (config.debug) consola.level = Number.MAX_SAFE_INTEGER;
  if (config.silent) consola.level = 0;
  const codeGenProcess = new CodeGenProcess(config);
  return await codeGenProcess.start();
}

export { generateTemplates } from "./commands/generate-templates/index.js";
export * as constants from "./constants.js";
