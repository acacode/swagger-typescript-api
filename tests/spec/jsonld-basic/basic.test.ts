/* eslint-disable */
/* tslint:disable */

import { generateApi } from "../../../src";
import { describe, expect, it } from "vitest";

describe("JSON-LD basic test", () => {
  it("should generate JSON-LD types correctly", async () => {
    const { files } = await generateApi({
      input: require.resolve("./schema.json"),
      generateClient: false,
      jsonLdOptions: {
        generateContext: true,
        generateUtils: true,
        strictTyping: false,
      },
    });

    expect(files).toMatchSnapshot();
  });

  it("should generate JSON-LD types with strict typing", async () => {
    const { files } = await generateApi({
      input: require.resolve("./schema.json"),
      generateClient: false,
      jsonLdOptions: {
        generateContext: true,
        generateUtils: true,
        strictTyping: true,
      },
    });

    expect(files).toMatchSnapshot();
  });

  it("should handle JSON-LD entities with context", async () => {
    const { files } = await generateApi({
      input: require.resolve("./schema.json"),
      generateClient: false,
      jsonLdOptions: {
        generateContext: true,
        generateUtils: true,
        entityPrefix: "JsonLd",
        contextSuffix: "Context",
      },
    });

    expect(files).toMatchSnapshot();
  });
});
