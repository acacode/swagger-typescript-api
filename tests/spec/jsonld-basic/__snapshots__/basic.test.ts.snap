// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`JSON-LD basic test > should generate JSON-LD types correctly 1`] = `
[
  {
    "fileContent": "

  /** JSON-LD Entity: Person */
  export interface Person {
  /**
   * type
   * Property: type
   */
    type?: string,
  /**
   * x-jsonld
   * Property: x-jsonld
   */
    "x-jsonld"?: boolean,
  /**
   * x-jsonld-context
   * Property: x-jsonld-context
   */
    "x-jsonld-context"?: {
  /**
   * name
   * Property: name
   */
    name?: string,
  /**
   * email
   * Property: email
   */
    email?: string,
  /**
   * birthDate
   * Property: birthDate
   */
    birthDate?: ,

},
  /**
   * x-jsonld-type
   * Property: x-jsonld-type
   */
    "x-jsonld-type"?: string,
  /**
   * properties
   * Property: properties
   */
    properties?: {
  /**
   * @context
   * JSON-LD context defining the meaning of terms
   */
    "@context"?: object,
  /**
   * @type
   * JSON-LD type identifier
   */
    "@type": string,
  /**
   * @id
   * JSON-LD identifier (IRI)
   */
    "@id"?: string,
  /**
   * name
   * Property: name
   */
    name?: Record<string,any>,
  /**
   * email
   * Property: email
   */
    email?: Record<string,any>,
  /**
   * birthDate
   * Property: birthDate
   */
    birthDate?: Record<string,any>,

},
  /**
   * required
   * Property: required
   */
    required?: (string)[],
}

  /** JSON-LD Entity: Organization */
  export interface Organization {
  /**
   * type
   * Property: type
   */
    type?: string,
  /**
   * x-jsonld
   * Property: x-jsonld
   */
    "x-jsonld"?: boolean,
  /**
   * x-jsonld-context
   * Property: x-jsonld-context
   */
    "x-jsonld-context"?: string,
  /**
   * x-jsonld-type
   * Property: x-jsonld-type
   */
    "x-jsonld-type"?: string,
  /**
   * properties
   * Property: properties
   */
    properties?: {
  /**
   * @context
   * JSON-LD context defining the meaning of terms
   */
    "@context"?: object,
  /**
   * @type
   * JSON-LD type identifier
   */
    "@type": string,
  /**
   * @id
   * JSON-LD identifier (IRI)
   */
    "@id"?: string,
  /**
   * name
   * Property: name
   */
    name?: Record<string,any>,
  /**
   * url
   * Property: url
   */
    url?: Record<string,any>,
  /**
   * employees
   * Property: employees
   */
    employees?: Record<string,any>,

},
  /**
   * required
   * Property: required
   */
    required?: (string)[],
}

  /** JSON-LD Entity: JsonLdEntity */
  export interface JsonLdEntity {
  /**
   * @context
   * JSON-LD context defining the meaning of terms
   */
    "@context"?: object,
  /**
   * @type
   * JSON-LD type identifier
   */
    "@type": string,
  /**
   * @id
   * JSON-LD identifier (IRI)
   */
    "@id"?: string,
}

",
    "fileExtension": ".ts",
    "fileName": "Api",
  },
]
`;

exports[`JSON-LD basic test > should generate JSON-LD types with strict typing 1`] = `
[
  {
    "fileContent": "

  /** JSON-LD Entity: Person */
  export interface Person {
  /**
   * type
   * Property: type
   */
    type?: string,
  /**
   * x-jsonld
   * Property: x-jsonld
   */
    "x-jsonld"?: boolean,
  /**
   * x-jsonld-context
   * Property: x-jsonld-context
   */
    "x-jsonld-context"?: {
  /**
   * name
   * Property: name
   */
    name?: string,
  /**
   * email
   * Property: email
   */
    email?: string,
  /**
   * birthDate
   * Property: birthDate
   */
    birthDate?: ,

},
  /**
   * x-jsonld-type
   * Property: x-jsonld-type
   */
    "x-jsonld-type"?: string,
  /**
   * properties
   * Property: properties
   */
    properties?: {
  /**
   * @context
   * JSON-LD context defining the meaning of terms
   */
    "@context"?: object,
  /**
   * @type
   * JSON-LD type identifier
   */
    "@type": string,
  /**
   * @id
   * JSON-LD identifier (IRI)
   */
    "@id"?: string,
  /**
   * name
   * Property: name
   */
    name?: Record<string,any>,
  /**
   * email
   * Property: email
   */
    email?: Record<string,any>,
  /**
   * birthDate
   * Property: birthDate
   */
    birthDate?: Record<string,any>,

},
  /**
   * required
   * Property: required
   */
    required?: (string)[],
}

  /** JSON-LD Entity: Organization */
  export interface Organization {
  /**
   * type
   * Property: type
   */
    type?: string,
  /**
   * x-jsonld
   * Property: x-jsonld
   */
    "x-jsonld"?: boolean,
  /**
   * x-jsonld-context
   * Property: x-jsonld-context
   */
    "x-jsonld-context"?: string,
  /**
   * x-jsonld-type
   * Property: x-jsonld-type
   */
    "x-jsonld-type"?: string,
  /**
   * properties
   * Property: properties
   */
    properties?: {
  /**
   * @context
   * JSON-LD context defining the meaning of terms
   */
    "@context"?: object,
  /**
   * @type
   * JSON-LD type identifier
   */
    "@type": string,
  /**
   * @id
   * JSON-LD identifier (IRI)
   */
    "@id"?: string,
  /**
   * name
   * Property: name
   */
    name?: Record<string,any>,
  /**
   * url
   * Property: url
   */
    url?: Record<string,any>,
  /**
   * employees
   * Property: employees
   */
    employees?: Record<string,any>,

},
  /**
   * required
   * Property: required
   */
    required?: (string)[],
}

  /** JSON-LD Entity: JsonLdEntity */
  export interface JsonLdEntity {
  /**
   * @context
   * JSON-LD context defining the meaning of terms
   */
    "@context"?: object,
  /**
   * @type
   * JSON-LD type identifier
   */
    "@type": string,
  /**
   * @id
   * JSON-LD identifier (IRI)
   */
    "@id"?: string,
}

",
    "fileExtension": ".ts",
    "fileName": "Api",
  },
]
`;

exports[`JSON-LD basic test > should handle JSON-LD entities with context 1`] = `
[
  {
    "fileContent": "

  /** JSON-LD Entity: Person */
  export interface Person {
  /**
   * type
   * Property: type
   */
    type?: string,
  /**
   * x-jsonld
   * Property: x-jsonld
   */
    "x-jsonld"?: boolean,
  /**
   * x-jsonld-context
   * Property: x-jsonld-context
   */
    "x-jsonld-context"?: {
  /**
   * name
   * Property: name
   */
    name?: string,
  /**
   * email
   * Property: email
   */
    email?: string,
  /**
   * birthDate
   * Property: birthDate
   */
    birthDate?: ,

},
  /**
   * x-jsonld-type
   * Property: x-jsonld-type
   */
    "x-jsonld-type"?: string,
  /**
   * properties
   * Property: properties
   */
    properties?: {
  /**
   * @context
   * JSON-LD context defining the meaning of terms
   */
    "@context"?: object,
  /**
   * @type
   * JSON-LD type identifier
   */
    "@type": string,
  /**
   * @id
   * JSON-LD identifier (IRI)
   */
    "@id"?: string,
  /**
   * name
   * Property: name
   */
    name?: Record<string,any>,
  /**
   * email
   * Property: email
   */
    email?: Record<string,any>,
  /**
   * birthDate
   * Property: birthDate
   */
    birthDate?: Record<string,any>,

},
  /**
   * required
   * Property: required
   */
    required?: (string)[],
}

  /** JSON-LD Entity: Organization */
  export interface Organization {
  /**
   * type
   * Property: type
   */
    type?: string,
  /**
   * x-jsonld
   * Property: x-jsonld
   */
    "x-jsonld"?: boolean,
  /**
   * x-jsonld-context
   * Property: x-jsonld-context
   */
    "x-jsonld-context"?: string,
  /**
   * x-jsonld-type
   * Property: x-jsonld-type
   */
    "x-jsonld-type"?: string,
  /**
   * properties
   * Property: properties
   */
    properties?: {
  /**
   * @context
   * JSON-LD context defining the meaning of terms
   */
    "@context"?: object,
  /**
   * @type
   * JSON-LD type identifier
   */
    "@type": string,
  /**
   * @id
   * JSON-LD identifier (IRI)
   */
    "@id"?: string,
  /**
   * name
   * Property: name
   */
    name?: Record<string,any>,
  /**
   * url
   * Property: url
   */
    url?: Record<string,any>,
  /**
   * employees
   * Property: employees
   */
    employees?: Record<string,any>,

},
  /**
   * required
   * Property: required
   */
    required?: (string)[],
}

  /** JSON-LD Entity: JsonLdEntity */
  export interface JsonLdEntity {
  /**
   * @context
   * JSON-LD context defining the meaning of terms
   */
    "@context"?: object,
  /**
   * @type
   * JSON-LD type identifier
   */
    "@type": string,
  /**
   * @id
   * JSON-LD identifier (IRI)
   */
    "@id"?: string,
}

",
    "fileExtension": ".ts",
    "fileName": "Api",
  },
]
`;
